packages/tartaros/src/lib.rs

```diff
@@ -0,0 +1,582 @@
+        //         if x < rect.width() as f64 - SCROLL_BAR_WIDTH as f64 {
+        //             state.scroll_bar_y_offset = -1.0;
+        //             return;
+        //         }
+        //         if y < scroll_bar_y || y > scroll_bar_y + scroll_bar_height as f64 {
+        //             state.scroll_bar_y_offset = (scroll_bar_height as f64) / 2.0;
+        //             state.offset = PyroConsole::calculate_offset(
+        //                 y - scroll_bar_height as f64 / 2.0,
+        //                 rect.height(),
+        //                 (state.lines.len() - LINES_VISIBLE) as u64,
+        //             );
+        //         } else {
+        //             state.scroll_bar_y_offset = y - scroll_bar_y;
+        //         }
+        //     })
+        // };
+
+        // _state
+        //     .borrow()
+        //     .canvas
+        //     .add_event_listener_with_callback("mousedown", closure.as_ref().unchecked_ref())
+        //     .expect("failed to add mousedown listener");
+
+        // closure.forget();
+
+        // let closure = {
+        //     let state = _state.clone();
+        //     Closure::<dyn FnMut(_)>::new(move |event: web_sys::MouseEvent| {
+        //         let mut state = state.borrow_mut();
+        //         if state.scroll_bar_y_offset < 0.0 {
+        //             return;
+        //         }
+        //         let y = event.client_y() - state.scroll_bar_y_offset as i32;
+        //         let rect = state.canvas.get_bounding_client_rect();
+        //         let y = y - rect.top() as i32;
+
+        //         // for debugging, just scroll anyway
+        //         // figure out what offset we need for the current y position -- the top pixel of the
+        //         // scroll bar should be at the current y position (approximately)
+        //         let offset = PyroConsole::calculate_offset(
+        //             y as f64,
+        //             rect.height(),
+        //             (state.lines.len() - LINES_VISIBLE) as u64,
+        //         );
+        //         state.offset = offset;
+        //     })
+        // };
+
+        // _state
+        //     .borrow()
+        //     .window
+        //     .add_event_listener_with_callback("mousemove", closure.as_ref().unchecked_ref())
+        //     .expect("failed to add mousemove listener");
+
+        // closure.forget();
+
+        // let closure = {
+        //     let state = _state.clone();
+        //     Closure::<dyn FnMut(_)>::new(move |_: web_sys::MouseEvent| {
+        //         let mut state = state.borrow_mut();
+        //         state.scroll_bar_y_offset = -1.0;
+        //     })
+        // };
+
+        // _state
+        //     .borrow()
+        //     .window
+        //     .add_event_listener_with_callback("mouseup", closure.as_ref().unchecked_ref())
+        //     .expect("failed to add mouseup listener");
+
+        // closure.forget();
```

Comment:
Do you plan on reusing any of this further on in this PR? Otherwise I think this massive comment should probably be removed before this gets approved

Reply:
oh yah this was all old migration code for when i was moving it off the ui thread into a web worker. will remove